cmd/web-handlers.go

```diff
@@ -497,7 +497,11 @@ func (web *webAPIHandlers) Upload(w http.ResponseWriter, r *http.Request) {
 	}
 
 	// Extract incoming metadata if any.
-	metadata := extractMetadataFromHeader(r.Header)
+	metadata, err := extractMetadataFromHeader(r.Header)
+	if err != nil {
+		writeErrorResponse(w, ErrInvalidMetadataDirective, r.URL)
```

Comment:
The error being returned here is not correct - `extractMetadataFromHeader` returns an `ErrInvalidArgument` but we return an error about metadata directive? It should be an internal server error with a message about an invalid condition being found in the server. An invalid canonical header cannot be caused by the user or an application, so it is not correct so send such an error back to the user.